cmd/cosign/cli/options/certextensions.go

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package options
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	sigs "github.com/sigstore/cosign/pkg/signature"
+)
+
+// CertExtensionOptions is the top level wrapper for the annotations.
+type CertExtensionOptions struct {
+	CertExtensions []string
+}
+
+var _ Interface = (*CertExtensionOptions)(nil)
+
+func (o *CertExtensionOptions) CertExtensionsMap() (sigs.CertExtensionsMap, error) {
+	ce := sigs.CertExtensionsMap{}
+	for _, a := range o.CertExtensions {
+		kv := strings.Split(a, "=")
+		if len(kv) != 2 {
+			return ce, fmt.Errorf("unable to parse cert extension: %s", a)
+		}
+		if ce.CertExtensions == nil {
+			ce.CertExtensions = map[string]string{}
+		}
+		ce.CertExtensions[kv[0]] = kv[1]
+	}
+	return ce, nil
+}
+
+// AddFlags implements Interface
+func (o *CertExtensionOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringSliceVarP(&o.CertExtensions, "cert-extensions", "", nil,
```

Comment:
I think we discussed this on the comments, but it'd be preferrable to match the existing style of a flag per extension instead of have a map. 

See https://github.com/sigstore/cosign/blob/aad7c5fcd6e5f854ae48a57e42919039e6d219c0/cmd/cosign/cli/options/certificate.go for existing flags. We should add flags like `certificate-github-ref`, etc